R/pkg/R/column.R

```diff
@@ -32,35 +34,65 @@ setOldClass("jobj")
 #' @export
 #' @note Column since 1.4.0
 setClass("Column",
-         slots = list(jc = "jobj"))
+         slots = list(jc = "jobj", df = "SparkDataFrameOrNull"))
 
 #' A set of operations working with SparkDataFrame columns
 #' @rdname columnfunctions
 #' @name columnfunctions
 NULL
-
-setMethod("initialize", "Column", function(.Object, jc) {
+setMethod("initialize", "Column", function(.Object, jc, df) {
   .Object@jc <- jc
+
+  # Some Column objects don't have any referencing DataFrame. In such case, df will be NULL.
+  if (missing(df)) {
+    df <- NULL
+  }
+  .Object@df <- df
   .Object
 })
 
+setMethod("show", signature = "Column", definition = function(object) {
```

Comment:
@falaki @felixcheung I remember discussing this issue in the past when working on PR 8913. Adding parameter `n` to `show()` brings extra complications as follows:

Since parameter `n` is not part of `show()` signature, I'll need to change the generic definition, adding `... ` as an additional parameter. This causes some inconsistencies:

```
> df$Sepal_Length
An object of class "Column"
Slot "jc":
Java ref type org.apache.spark.sql.Column id 18 

Slot "df":
SparkDataFrame[Sepal_Length:double, Sepal_Width:double, Petal_Length:double, Petal_Width:double, Species:string]
```

```
> show(df$Sepal_Length)
 [1] 5.1 4.9 4.7 4.6 5.0 5.4 4.6 5.0 4.4 4.9 5.4 4.8 4.8 4.3 5.8 5.7 5.4 5.1 5.7
[20] 5.1
```
In the first case, the method that's invoked is `showDefault()` which, in turn invokes `show()`, but since I have changed the definition of `show()`, it is dispatching `methods:::show()` as opposed to `SparkR:::show()`. To fix this, I'll have to make `showDefault()` a generic and re-implement it, which I'd rather not to avoid potential conflicts with base R functions.

My suggestion would be to have a global parameter for the number of rows/elements to show which is **exactly** what R does.